MYPYPATH not working with local -stubs package

This is a bug report. However, if the current behavior is not a bug, then this is request to improve documentation.

## Description

Mypy doesn't recognise my local library typing stub.

## Reproduction:

### Actual Python file

```python
import aiomysql
```

### Directory tree
```
pythonlibtypes/
└── aiomysql-stubs
    ├── __init__.pyi
    ├── connection.pyi
    ├── cursors.pyi
    ├── pool.pyi
    ├── py.typed
    └── utils.pyi
```

### Mypy command

```
MYPYPATH="/path/to/project/pythonlibtypes/" pipenv run -- mypy --config mypy.ini module_to_check
```

`mypy.ini`:

```
[mypy]
warn_redundant_casts = True
warn_unused_ignores = True
incremental = True
follow_imports = normal
```

## Actual behavior

```
path/to/file.py:1: error: Cannot find module named 'aiomysql'
```

## Expected behavior

Import succeeds, Mypy uses the stub 

## Versions

Python 3.6.5

mypy 0.620

--- a/docs/source/installed_packages.rst
+++ b/docs/source/installed_packages.rst
@@ -30,6 +30,9 @@ imports, namespace packages, and custom import hooks.
 If you do not want to use typed packages, use the ``--no-site-packages`` flag
 to disable searching.
 
+Note that stub-only packages cannot be used with ``MYPYPATH``. If you want mypy
+to find the package, it must be installed.
+
 Making PEP 561 compatible packages
 **********************************
 

--- a/docs/source/installed_packages.rst
+++ b/docs/source/installed_packages.rst
@@ -30,8 +30,10 @@ imports, namespace packages, and custom import hooks.
 If you do not want to use typed packages, use the ``--no-site-packages`` flag
 to disable searching.
 
-Note that stub-only packages cannot be used with ``MYPYPATH``. If you want mypy
-to find the package, it must be installed.
+Note that stub-only packages (defined in
+`PEP 561 <https://www.python.org/dev/peps/pep-0561/#stub-only-packages>`_)
+cannot be used with ``MYPYPATH``. If you want mypy to find the package, it must
+be installed.
 
 Making PEP 561 compatible packages
 **********************************

--- a/docs/source/installed_packages.rst
+++ b/docs/source/installed_packages.rst
@@ -33,7 +33,9 @@ to disable searching.
 Note that stub-only packages (defined in
 `PEP 561 <https://www.python.org/dev/peps/pep-0561/#stub-only-packages>`_)
 cannot be used with ``MYPYPATH``. If you want mypy to find the package, it must
-be installed.
+be installed. For a package ``foo``, the name of the stub-only package
+(``foo-stubs``) is not a legal package name, so mypy will not find it, unless
+it is installed.
 
 Making PEP 561 compatible packages
 **********************************

--- a/docs/source/running_mypy.rst
+++ b/docs/source/running_mypy.rst
@@ -306,7 +306,7 @@ This is computed from the following items:
 
 .. note::
 
-    You cannot point to a PEP 561 package via the MYPYPATH, they must be
+    You cannot point to a PEP 561 package via the MYPYPATH, it must be
     installed (see :ref:`PEP 561 support <installed-packages>`)
 
 For sources given on the command line, the path is adjusted by crawling

--- a/docs/source/running_mypy.rst
+++ b/docs/source/running_mypy.rst
@@ -304,6 +304,11 @@ This is computed from the following items:
 - The relevant directories of the
   `typeshed <https://github.com/python/typeshed>`_ repo.
 
+.. note::
+
+    You cannot point to a PEP 561 package via the MYPYPATH, they must be
+    installed (see :ref:`PEP 561 support <installed-packages>`)
+
 For sources given on the command line, the path is adjusted by crawling
 up from the given file or package to the nearest directory that does not
 contain an ``__init__.py`` or ``__init__.pyi`` file.

--- a/docs/source/stubs.rst
+++ b/docs/source/stubs.rst
@@ -27,7 +27,7 @@ Here is an overview of how to create a stub file:
 Use the normal Python file name conventions for modules, e.g. ``csv.pyi``
 for module ``csv``. Use a subdirectory with ``__init__.pyi`` for packages. Note
 that PEP 561 stub-only packages must be installed, and may not be pointed at
-through the ``MYPYPATH``.
+through the ``MYPYPATH`` (see :ref:`PEP 561 support <installed-packages>`).
 
 If a directory contains both a ``.py`` and a ``.pyi`` file for the
 same module, the ``.pyi`` file takes precedence. This way you can

--- a/docs/source/stubs.rst
+++ b/docs/source/stubs.rst
@@ -25,7 +25,9 @@ Here is an overview of how to create a stub file:
     $ export MYPYPATH=~/work/myproject/stubs
 
 Use the normal Python file name conventions for modules, e.g. ``csv.pyi``
-for module ``csv``. Use a subdirectory with ``__init__.pyi`` for packages.
+for module ``csv``. Use a subdirectory with ``__init__.pyi`` for packages. Note
+that PEP 561 stub-only packages must be installed, and may not be pointed at
+through the ``MYPYPATH``.
 
 If a directory contains both a ``.py`` and a ``.pyi`` file for the
 same module, the ``.pyi`` file takes precedence. This way you can

--- a/docs/source/stubs.rst
+++ b/docs/source/stubs.rst
@@ -26,8 +26,9 @@ Here is an overview of how to create a stub file:
 
 Use the normal Python file name conventions for modules, e.g. ``csv.pyi``
 for module ``csv``. Use a subdirectory with ``__init__.pyi`` for packages. Note
-that PEP 561 stub-only packages must be installed, and may not be pointed at
-through the ``MYPYPATH`` (see :ref:`PEP 561 support <installed-packages>`).
+that `PEP 561 <https://www.python.org/dev/peps/pep-0561/>`_ stub-only packages
+must be installed, and may not be pointed at through the ``MYPYPATH``
+(see :ref:`PEP 561 support <installed-packages>`).
 
 If a directory contains both a ``.py`` and a ``.pyi`` file for the
 same module, the ``.pyi`` file takes precedence. This way you can

